aten/src/ATen/native/cuda/DistanceKernel.cu

```diff
@@ -0,0 +1,217 @@
+#include "ATen/ATen.h"
+#include <THC/THCTensorMathReduce.cuh>
+#include <math.h>
+
+#include "ATen/native/Distance.h"
+
+
+namespace at { namespace native {
+
+namespace {
+
+static const int warp_size = 32;
```

Comment:
Avoid hard-coding the size of the warp in CUDA files. It's different, for example, on AMD GPUs.  Using a fixed number of threads per block is fine.

Reply:
Ahh, I didn't know `warpSize` was built in. I will switch to that.